designs/2019-enable-es2017-by-default/README.md

```diff
@@ -0,0 +1,138 @@
+- Start Date: 2019-02-28
+- RFC PR: https://github.com/eslint/rfcs/pull/16
+- Authors: Teddy Katz
+
+# Enable ES2017 parsing and globals by default
+
+## Summary
+
+This RFC proposes to:
+
+* Add an `es2017` environment which enables all ES2015 and ES2017 globals.
+* Enable the `es2017` environment by default.
+* Change the `es6` environment to no longer enable `parserOptions: { ecmaVersion: 6 }`.
+* Change the default value of `parserOptions.ecmaVersion` in `espree` from `5` to `2017`.
+
+## Motivation
+
+### Users expect to be able to use new JS features by default
+
+By default, ESLint enables only the globals and syntax features from the ES5 spec, typically raising a linting error of some kind if more recent features are used. However, a large fraction of ESLint users want to use more recent features in their codebases, for a variety of reasons:
+
+* Some users only need their code to run in environments that support ES2015 (e.g. Node, or modern browsers other than IE11)
+* Some users use build tools such as `babel` or `typescript` to compile their ES2015+ code down to ES5.
+
+Due to the increased prevalence of self-updating browsers, new JS features are available at a much faster rate than in the past, reducing the need for users to be aware of which browsers/environments support which features. As a result, new users are often confused by the fact that ESLint raises parsing errors by default for ES2015 code. This adds friction to the process of creating a config for the first time, and might delay or prevent users from using ESLint at all.
+
+The current behavior is "noisy by default" -- if ESLint isn't sure what behavior the user wants and its default gets things wrong, an ES5 default makes the error manifest as a false positive rather than a false negative. This has some significant advantages -- a user is much more likely to notice a false positive than a false negative, allowing them to correct the config error.
+
+However, at this point it appears that most of our users would expect ES6 parsing by default. As a result, the advantage of a false positive over a false negative is outweighed by the fact that making this change would be replace a very large number of false positives with a small number of false negatives. (Using similar reasoning, [eslint/eslint#11105](https://github.com/eslint/eslint/issues/11105) updated `eslint --init` to set `parserOptions.ecmaVersion` to `2018` by default.)
+
+### The `es6` environment composes poorly with other environments because it sets `parserOptions.ecmaVersion`
```

Comment:
I’m +1 on this as well (itd be nice to separate this, and the 2017 env, from the default change.

Reply:
This could theoretically be separated, but it would end up being a breaking change for many more users because people are relying on `env: { es6: true }` resulting in a `parserOptions.ecmaVersion` value of at least 2015.

Reply:
is that not also the case for a larger env value? ES2015 isn’t the latest version that added new globals.

Reply:
I think the only new globals after ES2015 have been `Atomics` and `SharedArrayBuffer`, and we still don't have an environment containing those. (Aside from this RFC, the issues discussed in https://github.com/eslint/eslint/issues/9109 haven't been resolved.)